manifests/init.pp

```diff
@@ -283,171 +232,123 @@
+  String $ks_url = $::pulp::params::ks_url,
+  String $default_login = $::pulp::params::default_login,
+  String $default_password = $::pulp::params::default_password,
+  Boolean $debugging_mode = $::pulp::params::debugging_mode,
+  Enum['CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET'] $log_level = $::pulp::params::log_level,
+  Optional[Stdlib::Absolutepath] $server_working_directory = $::pulp::params::server_working_directory,
+  Stdlib::Absolutepath $rsa_key = $::pulp::params::rsa_key,
+  Stdlib::Absolutepath $rsa_pub = $::pulp::params::rsa_pub,
+  Stdlib::Absolutepath $ca_cert = $::pulp::params::ca_cert,
+  Stdlib::Absolutepath $ca_key = $::pulp::params::ca_key,
+  Optional[Stdlib::Absolutepath] $https_cert = $::pulp::params::https_cert,
+  Optional[Stdlib::Absolutepath] $https_key = $::pulp::params::https_key,
+  Optional[Stdlib::Absolutepath] $https_chain = $::pulp::params::https_chain,
+  Variant[String, Boolean] $ssl_username = $::pulp::params::ssl_username,
+  Integer $user_cert_expiration = $::pulp::params::user_cert_expiration,
+  Integer $consumer_cert_expiration = $::pulp::params::consumer_cert_expiration,
+  Stdlib::Absolutepath $serial_number_path = $::pulp::params::serial_number_path,
+  Integer[-1] $consumer_history_lifetime = $::pulp::params::consumer_history_lifetime,
+  Boolean $oauth_enabled = $::pulp::params::oauth_enabled,
+  String $oauth_key = $::pulp::params::oauth_key,
+  String $oauth_secret = $::pulp::params::oauth_secret,
+  Integer[0] $max_keep_alive = $::pulp::params::max_keep_alive,
+  String $messaging_url = $::pulp::params::messaging_url,
+  Enum['qpid', 'rabbitmq'] $messaging_transport = $::pulp::params::messaging_transport,
+  Boolean $messaging_auth_enabled = $::pulp::params::messaging_auth_enabled,
+  Optional[Stdlib::Absolutepath] $messaging_ca_cert = $::pulp::params::messaging_ca_cert,
+  Optional[Stdlib::Absolutepath] $messaging_client_cert = $::pulp::params::messaging_client_cert,
+  String $messaging_topic_exchange = $::pulp::params::messaging_topic_exchange,
+  Boolean $messaging_event_notifications_enabled = $::pulp::params::messaging_event_notifications_enabled,
+  Optional[String] $messaging_event_notification_url = $::pulp::params::messaging_event_notification_url,
+  String $messaging_version = $::pulp::params::messaging_version,
+  String $broker_url = $::pulp::params::broker_url,
+  Boolean $broker_use_ssl = $::pulp::params::broker_use_ssl,
+  Optional[String] $tasks_login_method = $::pulp::params::tasks_login_method,
+  String $email_host = $::pulp::params::email_host,
+  Integer $email_port = $::pulp::params::email_port,
+  String $email_from = $::pulp::params::email_from,
+  Boolean $email_enabled = $::pulp::params::email_enabled,
+  Boolean $manage_squid = $::pulp::params::manage_squid,
+  Optional[String] $lazy_redirect_host = $::pulp::params::lazy_redirect_host,
+  Optional[Integer[0, 65535]] $lazy_redirect_port = $::pulp::params::lazy_redirect_port,
```

Comment:
Nitpick, technically ports you *connect to* should start at 1, IIRC `0` is only an indication when listening that you'll take any free port.